cpp_utils/src/cpp/time/time_utils.cpp

```diff
@@ -95,8 +103,12 @@ std::string timestamp_to_string(
         bool local_time /* = false */)
 {
     std::ostringstream ss;
-    const std::chrono::high_resolution_clock::time_point::duration duration = timestamp.time_since_epoch();
-    const time_t duration_seconds = std::chrono::duration_cast<std::chrono::seconds>(duration).count();
+    time_t duration_seconds = std::chrono::duration_cast<std::chrono::seconds>(timestamp.time_since_epoch()).count();
+
+    #if _EPROSIMA_WINDOWS_PLATFORM // In Windows std::gmtime does not support negative values nor values greater than 2^32
+    time_t max_value = std::numeric_limits<long>::max();
+    duration_seconds = std::max((time_t) 0, std::min(max_value, duration_seconds));
+    #endif // if _EPROSIMA_WINDOWS_PLATFORM
```

Comment:
That's exactly what I meant. However perhaps we should print a warning if clamping has been done.

Reply:
I think we should also be handling similar problems in `string_to_timestamp` and `date_to_timestamp`.